javalib/src/main/scala/java/net/URLDecoder.scala

```diff
@@ -0,0 +1,46 @@
+package java.net
+
+import java.io.UnsupportedEncodingException
+import java.nio.{CharBuffer, ByteBuffer}
+import java.nio.charset.Charset
+
+object URLDecoder {
+  def decode(s: String, enc: String): String = {
```

Comment:
Ideally, both `decode(String)` and `decode(String, String)` would call an internal `decodeImpl(String, Charset)` which takes the instance of `Charset`.